core-bundle/src/Resources/contao/drivers/DC_Table.php

```diff
@@ -1652,9 +1649,14 @@ public function deleteChilds($table, $id, &$delete)
 
 			if ($objDelete->numRows && !($GLOBALS['TL_DCA'][$v]['config']['doNotDeleteRecords'] ?? null) && \strlen($v))
 			{
-				foreach ($objDelete->fetchAllAssoc() as $row)
+				$rows = $objDelete->fetchAllAssoc();
+				$this->preloadCurrentRecords(array_column($rows, 'id'), $table);
```

Comment:
Feels like you should use `setCurrentRecordCache` since the data is already available from the database! Or maybe do that in the loop and don't "get" the current record at all?